internal/provider/resource_vault_cluster_test.go

```diff
@@ -3,229 +3,78 @@ package provider
 import (
 	"context"
 	"fmt"
-	"regexp"
 	"testing"
+	"time"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
 )
 
-const vaultCluster = `
-resource "hcp_vault_cluster" "test" {
-	cluster_id         = "test-vault-cluster"
-	hvn_id             = hcp_hvn.test.hvn_id
-	tier               = "dev"
+type inputT struct {
+	vaultClusterName           string
+	hvnName                    string
+	vaultClusterResourceName   string
+	vaultClusterDataSourceName string
+	adminTokenResourceName     string
+	tf                         string
+	cloudProvider              string
+	region                     string
+	tier                       string
+	updateTier1                string
+	updateTier2                string
+	publicEndpoint             string
 }
-`
 
-// sets public_endpoint to true and add metrics and audit log
-const updatedVaultClusterPublicAndMetricsAuditLog = `
-resource "hcp_vault_cluster" "test" {
-	cluster_id         = "test-vault-cluster"
-	hvn_id             = hcp_hvn.test.hvn_id
-	tier               = "standard_small"
-	public_endpoint    = true
-	metrics_config			   {
-		splunk_hecendpoint = "https://http-input-splunkcloud.com"
-		splunk_token =       "test"
-	}
-	audit_log_config 		    {
-		datadog_api_key = "test_datadog"
-		datadog_region  = "us1"
-	}
-	major_version_upgrade_config {
-		upgrade_type = "MANUAL"
-	}
-}
-`
-
-// changes tier
-const updatedVaultClusterTierAndMVUConfig = `
-resource "hcp_vault_cluster" "test" {
-	cluster_id         = "test-vault-cluster"
-	hvn_id             = hcp_hvn.test.hvn_id
-	tier               = "standard_small"
-	major_version_upgrade_config {
-		upgrade_type = "SCHEDULED"
-		maintenance_window_day = "WEDNESDAY"
-		maintenance_window_time = "WINDOW_12AM_4AM"
+// This includes tests against both the resource, the corresponding datasource, and the dependent admin token resource
+// to shorten testing time.
+func TestAccVaultClusterAzure(t *testing.T) {
+	azureTestInput := inputT{
+		vaultClusterName:           addTimestampSuffix("test-vault-cluster-azure-"),
```

Comment:
Alas this is just over the 36 character limit 😩 
```suggestion
		vaultClusterName:           addTimestampSuffix("test-vault-azure-"),
```

Reply:
Done